src/games/strategy/triplea/ai/fastAI/FastAI.java

```diff
@@ -9,14 +9,16 @@
  */
 public class FastAI extends ProAI {
 
+  // Odds estimator
+  private final static IOddsCalculator estimator = new FastOddsEstimator();
+
   public FastAI(final String name, final String type) {
     super(name, type);
   }
 
   @Override
-  public void initializeBattleCalculator() {
-    final IOddsCalculator calc = new FastOddsEstimator();
-    ProBattleUtils.setOddsCalculator(calc);
+  protected void initializeCalc() {
+    calc = new ProOddsCalculator(estimator);
```

Comment:
Thanks for pulling that up. Looks a factory class really should be used to create these objects. If the AI knows about the "calc", why doesn't it know which "type" it is too?

```
AIFactory.create(playerName, aiType );  // then the factory coudl know which calc to inject into which AI.
```
Waht if we create a new AI, that uses the same concurrrent calc or a fast cal. Would we have to copy paste the constructor code from Fast AI to that new AI? Would we be creating an inheritance heirarchy to avoid copy/paste? Sounds limited. IOC is a useful pattern.